src/module_6/controllers/cart.ts

```diff
@@ -41,13 +36,20 @@ const updateCartController = async (req: Request, res: Response) => {
     })
 }
 
-const clearCartController = async (req: Request, res: Response) => {
-    const userId = getXUserHeader(req);
+const deleteCartController = async (req: Request, res: Response) => {
+    try {
+        const isAdmin = req.user.role === Role.admin;
```

Comment:
I recommend create middleware which will check user role before it goes to controller 
I see you have such middleware))